deployment/src/main/java/org/wso2/testgrid/deployment/DeploymentUtil.java

```diff
@@ -38,21 +43,36 @@
     /**
      * Reads the deployment.json file and constructs the deployment object.
      *
-     * @param testPlanLocation location String of the test plan
+     * @param workspace location String of the test plan
      * @return the deployment information ObjectMapper
      * @throws TestGridDeployerException If reading the deployment.json file fails
      */
-    public static DeploymentCreationResult getDeploymentCreationResult(String testPlanLocation)
+    public static DeploymentCreationResult getDeploymentCreationResult(String workspace)
             throws TestGridDeployerException {
-
+        DeploymentCreationResult deploymentCreationResult = new DeploymentCreationResult();
+        List<Host> hosts = new ArrayList<>();
         ObjectMapper mapper = new ObjectMapper();
-        File file = new File(Paths.get(testPlanLocation, DeployerConstants.DEPLOYMENT_FILE).toString());
+        File file = new File(Paths.get(workspace, DeployerConstants.DEPLOYMENT_FILE).toString());
         try {
-            return mapper.readValue(file, DeploymentCreationResult.class);
+            List<Host> hostList = mapper.readValue(file, DeploymentCreationResult.class).getHosts();
+            for (Host host : hostList) {
+                Host serverHost = new Host();
+                serverHost.setIp(host.getIp());
+                serverHost.setLabel("serverHost");
+                for (Port port : host.getPorts()) {
+                    Host serverPort = new Host();
+                    serverPort.setIp(String.valueOf(port.getPortNumber()));
+                    serverPort.setLabel("serverPort");
+                    hosts.add(serverPort);
+                }
+                hosts.add(serverHost);
```

Comment:
It would be better to add an explanation explaining why we have to add **serverPort** to **hosts** instead of **serverHost**.

Reply:
Added in a0f5e7e